wvd-templates/Create and provision WVD host pool/mainTemplate.json

```diff
@@ -292,81 +357,170 @@
+            },
+            "metadata": {
+                "comments": "If the following conditional dependencies are not deployed, ARM will auto remove them from this dependsOn list"
+            },
+            "dependsOn": [
+                "[concat('Microsoft.Compute/images/', variables('imageName'))]",
+                "[concat('Microsoft.Network/networkInterfaces/', variables('rdshPrefix'), copyindex(), '-nic')]"
+            ],
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('rdshVmSize')]"
+                },
+                "availabilitySet": "[if(not(parameters('enablePersistentDesktop')), main.getAvailabilitySet(variables('rdshPrefix')), json('null'))]",
+                "osProfile": {
+                    "computerName": "[concat(variables('rdshPrefix'), copyindex())]",
+                    "adminUsername": "[variables('existingDomainUsername')]",
+                    "adminPassword": "[parameters('existingDomainPassword')]"
+                },
+                "storageProfile": {
+                    "imageReference": "[if(equals(variables('rdshImageSourceLower'), 'customvhd'), main.getVMImageReferenceCustomVHD(variables('imageName')), variables('VMImageReference')[variables('rdshImageSourceLower')])]",
+                    "osDisk": {
+                        "createOption": "FromImage",
+                        "managedDisk": {
+                            "storageAccountType": "[parameters('rdshVMDiskType')]"
+                        }
+                    }
+                },
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', concat(variables('rdshPrefix'), copyindex(), '-nic'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": false
+                    }
+                },
+                "licenseType": "Windows_Client"
+            }
+        },
+        {
+            "condition": "[not(variables('rdshManagedDisks'))]",
+            "apiVersion": "2018-10-01",
+            "type": "Microsoft.Compute/virtualMachines",
+            "name": "[concat(if(variables('rdshManagedDisks'), 'null', ''), variables('rdshPrefix'), copyindex())]",
+            "location": "[parameters('location')]",
+            "copy": {
+                "name": "rdsh-vm-loop-unmanaged",
+                "count": "[parameters('rdshNumberOfInstances')]"
+            },
+            "metadata": {
+                "comments": "If the following conditional dependencies are not deployed, ARM will auto remove them from this dependsOn list"
+            },
+            "dependsOn": [
+                "[concat('Microsoft.Compute/images/', variables('imageName'))]",
```

Comment:
@nakranimohit0 : Is this supposed to be depend on the image, even if it's being created with a blob (unmanaged) VHD file?

Reply:
Any resources inside dependsOn list that will not be deployed because of the condition, will be ignored (or removed from dependsOn list). See the comment right above dependsOn in line 472. But I need to add another condition in the image resource to address your other comment.

Reply:
@nakranimohit0 : Sorry, what I mean is that this "virtualMachine" block will only get created when the source is unmanaged disk and the admin selects unmanaged disk for the VM. That's why I don't think there should be a dependency on the image.

Reply:
I see what you mean, I'll remove it for unmanaged disk VM resource.